### PR TITLE
chore: reinforce upgrade warnings/errors

### PR DESCRIPTION
**Pull request by jackfrancis**
**Reason for Change**:

This PR clarifies known-not-working upgrade configurations, and prevents you from breaking your cluster by attempting an upgrade in a cluster configuration that will produce side-effects if touched by `aks-engine upgrade`.

**Issue Fixed**:

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [ ] No
- [ ] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [ ] Yes

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: test/e2e/test_cluster_configs/everything.json [hunk 1]
```diff
@@ -17,7 +17,7 @@
 					"networkPlugin": "azure",
 					"networkMode": "bridge",
 					"enableEncryptionWithExternalKms": true,
-					"useManagedIdentity": true,
+					"useManagedIdentity": false,
```
Comment (jackfrancis) [inline test/e2e/test_cluster_configs/everything.json:20]: Setting this to `false` so that we use a sane KMS config

File: cmd/upgrade.go [hunk 2]
```diff
@@ -170,6 +170,18 @@ func (uc *upgradeCmd) loadCluster() error {
...
+	// Ensure there aren't known-breaking API model configurations
+	if uc.containerService.Properties.MasterProfile.AvailabilityProfile == api.VirtualMachineScaleSets {
+		return errors.Errorf("Clusters with a VMSS control plane are not upgradable using `aks-engine upgrade`!")
+	}
+	if uc.containerService.Properties.OrchestratorProfile != nil &&
+		uc.containerService.Properties.OrchestratorProfile.KubernetesConfig != nil &&
+		to.Bool(uc.containerService.Properties.OrchestratorProfile.KubernetesConfig.EnableEncryptionWithExternalKms) &&
+		to.Bool(uc.containerService.Properties.OrchestratorProfile.KubernetesConfig.UseManagedIdentity) &&
+		uc.containerService.Properties.OrchestratorProfile.KubernetesConfig.UserAssignedID == "" {
+		return errors.Errorf("Clusters with enableEncryptionWithExternalKms=true and system-assigned identity are not upgradable using `aks-engine upgrade`!")
```
Comment (aramase) [inline cmd/upgrade.go:182]: 👍🏻

**Review by mboersma — APPROVED** [review]
/lgtm

**Pull request merged**